### PR TITLE
DM-46188: fix and test LSST[Com]Cam[Sim] pipelines with the step-tester

### PR DESCRIPTION
**Pull request by TallJimbo**
This doesn't guarantee that these pipelines would actually run, but it at least ensures they stay self-consistent.

### CODE REVIEW

File: tests/test_pipelines.py [hunk 1]
```diff
@@ -82,6 +82,32 @@
     "truth_summary",
 }
 
+# LSSTCam common inputs, in addition to COMMON_INPUTS
+LSSTCAM_INPUTS = {
+    "bfk",
+    "linearizer",
+    "defects",
+    "the_monster_20240904",
+    "gaia_dr3_20230707",
```
Comment (erykoff) [inline tests/test_pipelines.py:91]: I think that the astrometric refcat should be the monster as well.

Comment (TallJimbo) [reply tests/test_pipelines.py:91]: I wasn't sure if it was time to make that switch but I'm happy to do it.

Comment (erykoff) [reply tests/test_pipelines.py:91]: I just merged the fix to meas_algorithms so it's good to go.  Also we don't have gaia_dr3 on the summit.

File: tests/test_pipelines.py [hunk 2]
```diff
@@ -82,6 +82,32 @@
...
+}
+
+# LSSTComCam common inputs, in addition to COMMON_INPUTS
+LSSTCOMCAM_INPUTS = {
+    "bfk",
+    "ptc",
+    "linearizer",
+    "defects",
+    "the_monster_20240904",
+    "gaia_dr3_20230707",
```
Comment (erykoff) [inline tests/test_pipelines.py:101]: This should be just the_monster as well.

File: tests/test_pipelines.py [hunk 3]
```diff
@@ -220,6 +246,24 @@
...
+    "calexp", "visitSummary"
+}
+
+# All refcats used by any pipelines, for inclusion as the initial_dataset_types
+# argument to PipelineStepTester; the intent is just to allow the 'skypix'
+# dimension placeholder used in these connections to be resolved.
+REFCATS = [
+    ("ps1_pv3_3pi_20170110", {"htm7"}, "SimpleCatalog", False),
+    ("gaia_dr2_20200414", {"htm7"}, "SimpleCatalog", False),
+    ("gaia_dr3_20230707", {"htm7"}, "SimpleCatalog", False),
```
Comment (erykoff) [inline tests/test_pipelines.py:260]: I think it's okay to have this here even though we don't have all these refcats on the summit.

Comment (erykoff) [reply tests/test_pipelines.py:260]: Though do we need a subset that is on the summit and a fuller set that is at DFs and make sure that quicklook uses only summit refcats?

**Review by erykoff — APPROVED** [review]
Looks good, but some questions about summit refcats and quicklook pipelines.

File: tests/test_pipelines.py (hunk 3 above)
Comment (TallJimbo) [reply tests/test_pipelines.py:261]: This is just a convenience list of all the refcats anywhere (note that it includes simulated ones).  The real check on which refcat is used in a pipeline is the variables used to supply `expected_inputs` to the tester.

Comment (erykoff) [reply tests/test_pipelines.py:261]: Ah!  I see.  Then my comment above that gaia_dr3 should be removed from the lsst inputs.

**Review by erykoff — APPROVED** [review]

**Pull request merged**